python_bindings/CMakeLists.txt

```diff
@@ -0,0 +1,247 @@
+find_package(pybind11 REQUIRED)
+
+if (APPLE)
+  set(HALIDE_PY_LINK_FLAG "-Wl,-undefined,dynamic_lookup")
+elseif (NOT MSVC)
+  set(HALIDE_PY_LINK_FLAG "-rdynamic")
+endif()
+
+if (NOT HALIDE_ENABLE_RTTI)
+  if (NOT MSVC)
+    set(HALIDE_PY_RTTI_FLAG "-fno-rtti")
+  else()
+    set(HALIDE_PY_RTTI_FLAG "/GR-")
+  endif()
+endif()
+
+pybind11_add_module(halide_py
+                    src/PyArgument.cpp            
+                    src/PyBoundaryConditions.cpp  
+                    src/PyBuffer.cpp              
+                    src/PyConciseCasts.cpp        
+                    src/PyDerivative.cpp          
+                    src/PyEnums.cpp               
+                    src/PyError.cpp               
+                    src/PyExpr.cpp                
+                    src/PyExternFuncArgument.cpp  
+                    src/PyFunc.cpp                
+                    src/PyFuncRef.cpp             
+                    src/PyHalide.cpp              
+                    src/PyIROperator.cpp        
+                    src/PyImageParam.cpp        
+                    src/PyInlineReductions.cpp  
+                    src/PyLambda.cpp            
+                    src/PyLoopLevel.cpp         
+                    src/PyMachineParams.cpp     
+                    src/PyModule.cpp    
+                    src/PyParam.cpp     
+                    src/PyPipeline.cpp  
+                    src/PyRDom.cpp      
+                    src/PyStage.cpp
+                    src/PyTarget.cpp
+                    src/PyTuple.cpp
+                    src/PyType.cpp
+                    src/PyVar.cpp
+                    src/PyVarOrRVar.cpp)
+set_target_properties(halide_py PROPERTIES LIBRARY_OUTPUT_NAME "halide"
+                                           LINK_FLAGS "${HALIDE_PY_LINK_FLAG}")
+target_link_libraries(halide_py PRIVATE "${HALIDE_COMPILER_LIB}")
+target_include_directories(halide_py PRIVATE "${HALIDE_INCLUDE_DIR}" "${HALIDE_TOOLS_DIR}")
+target_compile_options(halide_py PRIVATE $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-cast-qual>
+                                         $<$<CXX_COMPILER_ID:GNU>:-Wno-suggest-override>)
+
+install(TARGETS halide_py LIBRARY DESTINATION bin)
+
+_halide_library_runtime(host HALIDE_TARGET)
+
```

Comment:
This should be in its own CMakeLists.txt in `src` and then `add_subdirectory(src)` here. This issue repeats below with the tests.